tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/team/mailboxes/TeamMailboxExtensionBehaviourImpl.scala

```diff
@@ -1,12 +1,12 @@
 package com.linagora.tmail.james.jmap.team.mailboxes
 
 import com.linagora.tmail.james.jmap.contact.{ContactFields, EmailAddressContactSearchEngine}
-import com.linagora.tmail.team.TeamMailbox
+import com.linagora.tmail.team.{TeamMailbox, TeamMailboxExtensionBehaviour}
 import javax.inject.Inject
 import org.reactivestreams.Publisher
 import reactor.core.scala.publisher.SMono
 
-class TeamMailboxCallbackUtils @Inject()(contactSearchEngine: EmailAddressContactSearchEngine) {
+class TeamMailboxExtensionBehaviourImpl @Inject()(contactSearchEngine: EmailAddressContactSearchEngine) extends TeamMailboxExtensionBehaviour {
```

Comment:
Thank you for the interface idea, that helped me solve the circular dependency issue.

I guess you want this to be even more modular (multiple behavior implementation e.g auto-complete, ...).
But at the moment there is limited extension behavior, so now I would prefer keeping this simple by aggregating behaviors into a single impl class.

If that is ok to you?